lib/IDE/TypeReconstruction.cpp

```diff
@@ -1186,6 +1186,37 @@ static void VisitNodeFunction(
       VisitNode(ast, *pos, decl_scope_result);
       break;
 
+    case Demangle::Node::Kind::PrivateDeclName: {
+      if (child->getNumChildren() != 2 || !child->getChild(0)->hasText() ||
+          !child->getChild(1)->hasText()) {
+        result._error =
+              "unable to retrieve content for Node::Kind::PrivateDeclName";
+        break;
+      }
+
+      Demangle::NodePointer priv_decl_id_node(child->getChild(0));
+      Demangle::NodePointer id_node(child->getChild(1));
+
+      FindNamedDecls(ast, ast->getIdentifier(id_node->getText()),
+                     decl_scope_result, priv_decl_id_node->getText().str());
+      if (decl_scope_result._decls.empty()) {
+        result._error = stringWithFormat(
+            "demangled identifier %s could not be found by name lookup",
+            id_node->getText().str().c_str());
+        break;
+      }
```

Comment:
Perhaps add a comment what is being done here?

Reply:
I used the helper, so this should be fairly self-explanatory now. Please let me know if you want me to add something else.